compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -1091,3 +1103,37 @@ fn try_instance_mir<'tcx>(
     }
     Ok(tcx.instance_mir(instance))
 }
+
+fn body_is_forwarder(body: &Body<'_>) -> bool {
```

Comment:
may be useful to look into methods that only forward to other methods, but may trigger autoderef through `Deref` impls or other things that can cause this method to return `false`